tests/test_templates.py

```diff
@@ -12,106 +12,307 @@
+                                 [[1.87210909, 3.59695024], [1.42759279, 3.84330071]],
+                                 [[0.00389139,  0.05125553], [-0.12120044,  0.03111934]]
+                                 ]),
+            (Interferometer, [[2.33312851], [3.49488327], [0.9868003, 1.58798724]])
+            ]
+
+
+class TestIntegrationInitialization:
+    """Tests integration with the parameter initialization functions from pennylane.init"""
+
+    @pytest.mark.parametrize(fixture, qubit_func)
+    def test_integration_qubit_init(self, template, inpts, qubit_device, n_subsystems, n_layers):
+        """Checks parameter initialization compatible with continuous-variable templates."""
+
+        inp = inpts(n_layers=n_layers, n_wires=n_subsystems)
+
+        @qml.qnode(qubit_device)
+        def circuit(inp_):
+            template(*inp_, wires=range(n_subsystems))
             return qml.expval(qml.Identity(0))
 
-        circuit(weights=p)
+        circuit(inp)
 
-    @pytest.mark.parametrize('parfun', [cvqnn_layer_uniform, cvqnn_layer_normal])
-    def test_integration_cvqnn_layer(self, parfun, gaussian_device, n_subsystems):
-        """Checks that parameters generated by methods from pennylane.init integrate
-        with pennnylane.templates.layers.CVNeuralNetLayer()."""
+    @pytest.mark.parametrize(fixture, cv_func)
+    def test_integration_cv_init(self, template, inpts, gaussian_device, n_subsystems, n_layers):
+        """Checks parameter initialization compatible with continuous-variable templates."""
 
-        p = parfun(n_wires=n_subsystems)
+        inp = inpts(n_layers=n_layers, n_wires=n_subsystems)
 
         @qml.qnode(gaussian_device)
-        def circuit(weights):
-            CVNeuralNetLayer(*weights, wires=range(n_subsystems))
+        def circuit(inp_):
+            template(*inp_, wires=range(n_subsystems))
             return qml.expval(qml.Identity(0))
 
-        circuit(weights=p)
+        circuit(inp)
 
-    @pytest.mark.parametrize('parfun', [strong_ent_layers_uniform, strong_ent_layers_normal])
-    def test_integration_stronglyentangling_layers(self, parfun, qubit_device, n_subsystems, n_layers):
-        """Checks that parameters generated by methods from pennylane.init integrate
-        with pennnylane.templates.layers.StronglyEntanglingLayers()."""
 
-        p = parfun(n_layers=n_layers, n_wires=n_subsystems)
+class TestIntegrationCircuit:
+    """Tests the integration of templates into circuits using the NumPy interface. """
 
-        @qml.qnode(qubit_device)
-        def circuit(weights):
-            StronglyEntanglingLayers(*weights, wires=range(n_subsystems))
-            return qml.expval(qml.Identity(0))
+    @pytest.mark.parametrize(fixture, qubit_const)
+    def test_integration_qubit_positional_arg(self, template, inpts):
+        """Checks integration of qubit templates using positional arguments to qnode."""
 
-        circuit(weights=p)
+        inpts = [np.array(i) for i in inpts]
```

Comment:
How come you had to do this? Were the tests breaking otherwise? :thinking:

Reply:
Not at all, I just wanted to emphasize that this is the input for the numpy inteface. 

Could also test lists.

Reply:
I checked, and lists are actually not supported by all templates. This will be addressed in a separate PR tackling signatures.